setup.py

```diff
@@ -34,35 +34,41 @@
     if os.path.exists('MANIFEST'):
         os.remove('MANIFEST')
 
-    setup(name=DISTNAME,
-          maintainer=MAINTAINER,
-          include_package_data=True,
-          maintainer_email=MAINTAINER_EMAIL,
-          description=DESCRIPTION,
-          license=LICENSE,
-          url=URL,
-          version=VERSION,
-          download_url=DOWNLOAD_URL,
-          long_description=open('README.md').read(),
-          zip_safe=False,  # the package can run out of an .egg file
-          classifiers=['Intended Audience :: Science/Research',
-                       'Intended Audience :: Developers',
-                       'License :: OSI Approved',
-                       'Programming Language :: Python',
-                       'Topic :: Software Development',
-                       'Topic :: Scientific/Engineering',
-                       'Operating System :: Microsoft :: Windows',
-                       'Operating System :: POSIX',
-                       'Operating System :: Unix',
-                       'Operating System :: MacOS'],
-          platforms='any',
-          packages=find_packages(),
-          entry_points={
-              'console_scripts': [
-                  'ramp_test_submission='
-                  'rampwf.utils.command_line:ramp_test_submission',
-                  'ramp_test_notebook='
-                  'rampwf.utils.command_line:ramp_test_notebook',
-                  'ramp_convert_notebook='
-                  'rampwf.utils.command_line:ramp_convert_notebook',
-              ]})
+    setup(
+        name=DISTNAME,
+        maintainer=MAINTAINER,
+        include_package_data=True,
+        maintainer_email=MAINTAINER_EMAIL,
+        description=DESCRIPTION,
+        license=LICENSE,
+        url=URL,
+        version=VERSION,
+        download_url=DOWNLOAD_URL,
+        long_description=open('README.md').read(),
+        zip_safe=False,  # the package can run out of an .egg file
+        classifiers=[
+            'Intended Audience :: Science/Research',
+            'Intended Audience :: Developers',
+            'License :: OSI Approved',
+            'Programming Language :: Python',
+            'Topic :: Software Development',
+            'Topic :: Scientific/Engineering',
+            'Operating System :: Microsoft :: Windows',
+            'Operating System :: POSIX',
+            'Operating System :: Unix',
+            'Operating System :: MacOS'],
+        install_requires=[
+            'numpy>=1.13',
+            'scipy>=0.19',
```

Comment:
scipy is not needed I think?

Reply:
indeed, but scikit-learn expects it and it is not installed as dependency

Reply:
Maybe better to read from `requrements.txt` here? If you install scikit-learn with conda, scipy will get installed... Also, joblib is imported but is not in the requirements...

Reply:
I agree its a duplication. But I don't know what is best regarding pypi.

Concerning joblib, I moved the imports to get rid of the dependency.

Reply:
let's leave them separate, the small duplication is not that bad. There is some slight ideological difference between requirements.txt (deploy specific versions) and install_requires (specify minimum dependencies the deployed environment should adhere to). See https://packaging.python.org/discussions/install-requires-vs-requirements/